dist/pom.xml

```diff
@@ -93,6 +93,14 @@
                 </included_buildvers>
             </properties>
         </profile>
+        <profile>
+            <id>jdk17</id>
```

Comment:
I'm confused why there's a jdk17 vs. a jdk17-build profile, and it would be nice if there was just a single profile for JDK17.  If this is really desired, please add comments to these profiles explaining when one would be used vs. the other.  jdk17-build is documented in CONTRIBUTING, but jdk17 is not.

Reply:
the profiles in dist/pom were only there for version-def usage in https://github.com/NVIDIA/spark-rapids/blob/branch-23.06/jenkins/version-def.sh#L55 before. If we need to combine them together, we will need to investigate if any other place is still depends on this one. 

I will add the TODO here